### PR TITLE
[ABC-472] Not Visible Mesh Import Bug

### PR DESCRIPTION
**Pull request by jun-loke**
## Purpose of this PR

**Ticket/Jira #:** ABC-472

Initialize the curves renderer on create instead of only when the component is visible 

When we import a node with the visibility property off, we add that as a child GO that is not active. Therefore the OnEnable functions of that component doesn't get called. As a result, when the initialization code is in the OnEnable, it was not getting run for the not visible nodes. This means that when we tried to add the curve renderer, the shared material was not set, causing the bug. So I just moved it out. 
To note, the init code needs to remain in the onEnable because when we toggle the game object's Active, it needs to run again otherwise the shared mesh gets set and unset

## Testing

**Functional Testing status:**

**Performance Testing status:**

## Overall Product Risks

**Complexity:**
Minimal / **Low** / Medium / High

**Halo Effect:**
Minimal /  **Low** / Medium / High

## Documentation & UX Writing

| User facing text to review |  | Details |
| :--- | :--- | :--- |
| User interface | N |  |
| Public API docs |N  |  |
| Changelog | Y |  |

| Documentation halo effect |  | Jira link |
| :--- | :--- | :--- |
| User manual |  |  |
| Other docs |  |  |

## Additional information

**Note to reviewers:**

**Reminder:**

- [x] Add entry in CHANGELOG.md

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.formats.alembic?pullRequest=568) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.formats.alembic?pullRequest=568) before we can accept your contribution.<br/><hr/>**noreply@unity3d.com** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.formats.alembic?pullRequest=568) it.</sub>

File: com.unity.formats.alembic/Runtime/Scripts/Importer/Behaviours/AlembicCurvesRenderer.cs [hunk 1]
```diff
@@ -26,19 +26,20 @@ public class AlembicCurvesRenderer : MonoBehaviour
...
-        void OnEnable()
+        internal void Init()
         {
             curves = GetComponent<AlembicCurves>();
             //curves.OnUpdate += UpdateMesh;
 
-            mesh = new Mesh { hideFlags = HideFlags.DontSave };
+            if (mesh == null)
+                mesh = new Mesh { hideFlags = HideFlags.DontSave };
+
```
Comment (thomas-tu) [inline com.unity.formats.alembic/Runtime/Scripts/Importer/Behaviours/AlembicCurvesRenderer.cs:42]: Do we need to re-execute the whole scope after a first init?

Comment (thomas-tu) [reply com.unity.formats.alembic/Runtime/Scripts/Importer/Behaviours/AlembicCurvesRenderer.cs:42]: Forget it, I missed the `nDisable` definition.

File: com.unity.formats.alembic/Tests/Editor/EditorTests.cs [hunk 2]
```diff
@@ -83,6 +83,32 @@ public IEnumerator AddCurveRenderingSettingIsObeyed([Values] bool addCurve)
...
+            importer.SaveAndReimport();
+
+            // instantiate prefab
+            var asset = AssetDatabase.LoadAssetAtPath<GameObject>(path);
+            var inst = PrefabUtility.InstantiatePrefab(asset) as GameObject;
+            var player = inst.GetComponent<AlembicStreamPlayer>();
+            Assume.That(player != null);
+
+            // Assert
+            var curvesRenderer = inst.GetComponentInChildren<AlembicCurvesRenderer>(true);
```
Comment (arejayelle) [inline com.unity.formats.alembic/Tests/Editor/EditorTests.cs:108]: `true` is to look for the component in the inactive children

**Review by cguer — APPROVED** [review]
QA approved 👍!

Tested on:
- 2023.2.0b15 (Windows only)

**Review by thomas-tu — APPROVED** [review]
LGTM!

**Pull request merged**